src/devices/Mpu9250/README.md

```diff
@@ -0,0 +1,185 @@
+
+## Uage
+
+You can find an example in the [sample](./samples/Mpu9250.sample.cs) directory. Usage is straightforward including the possibility to have a calibration for all sub sensors.
+
+```csharp
+var mpui2CConnectionSettingmpus = new I2cConnectionSettings(1, Mpu9250.DefaultI2cAddress);
+Mpu9250 mpu9250 = new Mpu9250(I2cDevice.Create(mpui2CConnectionSettingmpus));
+var gyro = mpu9250.GetGyroscope();
+Console.WriteLine($"Gyro X = {gyro.X, 15}");
+Console.WriteLine($"Gyro Y = {gyro.Y, 15}");
+Console.WriteLine($"Gyro Z = {gyro.Z, 15}");
+var acc = mpu9250.GetAccelerometer();
+Console.WriteLine($"Acc X = {acc.X, 15}");
+Console.WriteLine($"Acc Y = {acc.Y, 15}");
+Console.WriteLine($"Acc Z = {acc.Z, 15}");
+Console.WriteLine($"Temp = {mpu9250.Temperature.Celsius.ToString("0.00")} °C");
+var magne = mpu9250.ReadMagnetometer(true);
+Console.WriteLine($"Mag X = {magne.X, 15}");
+Console.WriteLine($"Mag Y = {magne.Y, 15}");
+Console.WriteLine($"Mag Z = {magne.Z, 15}");
+```
+
+## Self-test
+
+A self-test is available for the gyroscope and the accelerometer.
+
+```csharp
+var resSelfTest = mpu9250.RunGyroscopeAccelerometerSelfTest();
+Console.WriteLine($"Self test:");
+Console.WriteLine($"Gyro X = {resSelfTest.Item1.X} vs >0.005");
+Console.WriteLine($"Gyro Y = {resSelfTest.Item1.Y} vs >0.005");
+Console.WriteLine($"Gyro Z = {resSelfTest.Item1.Z} vs >0.005");
+Console.WriteLine($"Acc X = {resSelfTest.Item2.X} vs >0.005 & <0.015");
+Console.WriteLine($"Acc Y = {resSelfTest.Item2.Y} vs >0.005 & <0.015");
+Console.WriteLine($"Acc Z = {resSelfTest.Item2.Z} vs >0.005 & <0.015");
+```
+
+The returned data are the raw data and allows you to estimate the quality of the test. The first item of the tuple is the gyroscope and the second one the accelerometer.
+
+No self-test is available for the magnetometer.
+
+## Calibration and bias
+
+You can calibrate the Gyroscope and the Accelerometer at the same time. This action is as well correcting the registers directly in the MPU9250 chip. Those data are lost in case of power stop but stays in case of soft reset.
+
+```csharp
+Console.WriteLine("Running Gyroscope and Accelerometer calibration");
+mpu9250.CalibrateGyroscopeAccelerometer();
+Console.WriteLine("Calibration results:");
+Console.WriteLine($"Gyro X bias = {mpu9250.GyroscopeBias.X}");
+Console.WriteLine($"Gyro Y bias = {mpu9250.GyroscopeBias.Y}");
+Console.WriteLine($"Gyro Z bias = {mpu9250.GyroscopeBias.Z}");
+Console.WriteLine($"Acc X bias = {mpu9250.AccelerometerBias.X}");
+Console.WriteLine($"Acc Y bias = {mpu9250.AccelerometerBias.Y}");
+Console.WriteLine($"Acc Z bias = {mpu9250.AccelerometerBias.Z}");
+```
+
+Calibration is as well available for the magnetometer (the AK8963). For this sensor, no correction is done.
+
+```csharp
+Console.WriteLine("Magnetometer calibration is taking couple of seconds, please be patient!");
```

Comment:
should the docs and this text say to move the device around for better results? I've found with other magnetometer just leaving it is usually not enough

Reply:
Yes, good point. for this one, idea is not to move it at all. It should be far from phones, magnets, etc. I'll adjust.